buildSrc/src/main/groovy/org/opencms/GwtCompiler.groovy

```diff
@@ -0,0 +1,203 @@
+package org.opencms
+
+import org.gradle.api.tasks.*
+
+/**
+ * Invokes the GWT compiler on the OpenCms-core sources.
+ * <p>
+ * It requires the gradle source sets:
+ * <ul>
+ *     <li>{@link #sourceSetMain main}</li>
+ *     <li>{@link #sourceSetModules modules}</li>
+ *     <li>{@link #sourceSetGwt gwt}</li>
+ * </ul>
+ * During development, use {@link #draft} mode to improve compilation speed.
+ *
+ * @since 2017-07
+ */
+class GwtCompiler extends JavaExec {
+    @Input
+    String opencmsModuleName
+
+    /**
+     * GWT module name (e.g.: "{@code org.opencms.ugc.Ugc}").
+     * <p>
+     * In GWT Jargon, a module is an individual unit of GWT configuration. It bundles
+     * together all the configuration settings that the GWT project needs (e.g.: inherited
+     * modules, entry point application class name, source path entries, etc. ).
+     * <p>
+     * Modules are defined in XML and placed into the project package hierarchy. It is
+     * recommended that modules appear in the root package of <a href="http://www.gwtproject.org/doc/latest/DevGuideOrganizingProjects.html#DevGuideDirectoriesPackageConventions">the standard project layout</a>.
+     */
+    @Input
+    String gwtModuleName
+
+    /**
+     * Create only a draft version. Use only during development to achieve maximum compilation
+     * speed.
+     * <p>
+     * Default value: {@code false}
+     */
+    @Input
+    boolean draft = false
+
+    private SourceSet sourceSetMain
+
+    private SourceSet sourceSetModules
+
+    private SourceSet sourceSetGwt
+
+    /** The directory into which deployable output files will be written (defaults to 'war') */
+    private String gwtWarDir
```

Comment:
This variable was originally named `buildDir`. I have renamed it to `gwtWarDir` to avoid hiding `project.buildDir`